### PR TITLE
feature/sketch kore java interface

### PR DESCRIPTION
**Pull request by cos**
**please review**

Mostly sketches of the main data structures. We'll have to use them for a while to make sure everything is in order.

Also, once the interface stabilizes a bit (after we use it for a bit) I will document how to use it on a wiki page. For now, consider the examples as being the documentation.

### CODE REVIEW

**Comment by cos** [comment]
Jenkins: test this please

**Comment by osa1** [comment]
Indentations are different from rest of the framework.

**Comment by radumereuta** [comment]
I agree with @osa1 here. https://github.com/kframework/k/wiki/Style-Guidelines

**Comment by cos** [comment]
we're using a different language, so the style can be different. I'm using the official Scala style guide:
http://docs.scala-lang.org/style/indentation.html

**Comment by osa1** [comment]
@cos there some Java files in this patch.

**Comment by cos** [comment]
@osa1, thanks. fixed the indentation.

**Comment by cos** [comment]
@dwightguth, @osa1, please have a look.

**Comment by cos** [comment]
also, the the description in the first comment.

File: kore/pom.xml [hunk 1]
```diff
@@ -24,6 +24,12 @@
       <artifactId>scala-library</artifactId>
       <version>${scala.version}</version>
     </dependency>
+    <dependency>
```
Comment (dwightguth) [inline kore/pom.xml]: this pom should already have access to junit 4.11. You shouldn't need this.

**Comment by dwightguth** [comment]
one minor comment plus you should also merge against master because this pull request still references the diff that we merged already with sbt.

**Comment by cos** [comment]
removed the redundant dependency and rebased.

**Comment by dwightguth** [comment]
Merge if tests pass.

**Pull request merged**